packages/server/src/auth/newuser.ts

```diff
@@ -9,14 +9,17 @@ import { sendOutcome } from '../fhir/outcomes';
 import { systemRepo } from '../fhir/repo';
 import { globalLogger } from '../logger';
 import { getUserByEmailInProject, getUserByEmailWithoutProject, tryLogin } from '../oauth/utils';
-import { bcryptHashPassword, getProjectByRecaptchaSiteKey, verifyRecaptcha } from './utils';
 import { makeValidationMiddleware } from '../util/validator';
+import { bcryptHashPassword, getProjectByRecaptchaSiteKey, verifyRecaptcha } from './utils';
 
 export const newUserValidator = makeValidationMiddleware([
-  body('firstName').notEmpty().withMessage('First name is required'),
-  body('lastName').notEmpty().withMessage('Last name is required'),
-  body('email').isEmail().withMessage('Valid email address is required'),
-  body('password').isLength({ min: 8 }).withMessage('Password must be at least 8 characters'),
+  body('firstName').isLength({ min: 1, max: 72 }).withMessage('First name must be between 8 and 72 characters'),
+  body('lastName').isLength({ min: 1, max: 72 }).withMessage('Last name must be between 8 and 72 characters'),
+  body('email')
+    .isEmail()
+    .isLength({ min: 3, max: 72 })
+    .withMessage('Email must be valid email address between 3 and 72 characters'),
+  body('password').isLength({ min: 8, max: 72 }).withMessage('Password must be between 8 and 72 characters'),
```

Comment:
We should be wary of the non-ASCII case (ie. non-English), this will break when passwords are 2 or 4 byte character glyphs

Not sure how we want to deal with that yet, but maybe this would be a good thing to consider

Reply:
Yeah, this is a fair callout.  I wouldn't say it "breaks", but the behavior does get weird.

<img width="163" alt="image" src="https://github.com/medplum/medplum/assets/749094/b6a36dc5-427f-4dcc-95d5-c723dbe8cff1">

In general, I'm comfortable with that.  If users start sending emojis or special characters, then the counts could get weird.

I would not want to block non-ASCII passwords, because that's kinda unfriendly to international users.

Reply:
It breaks in the sense that it truncates the password, which I would think is more than just weird behavior. If the max was 25 let's say, and the user entered a password in Russian like this:
`"здравствуйте, Медплум"`

This is what happens:

<img width="471" alt="Screenshot 2023-11-18 at 1 55 21 PM" src="https://github.com/medplum/medplum/assets/1592008/a326e286-c510-44ec-92ad-fbd8c4d6e694">

Assuming that `bcrypt` truncated at 25 instead of 72, this is what the password would actually be in practice:
`"здравствуйте"`

Because of this:

<img width="455" alt="Screenshot 2023-11-18 at 1 55 46 PM" src="https://github.com/medplum/medplum/assets/1592008/f699c0fb-d10e-4b51-9f47-e12388ff4c8d">

Which leads to this becoming the de facto password as `bcrypt` sees it:

<img width="504" alt="Screenshot 2023-11-18 at 1 58 05 PM" src="https://github.com/medplum/medplum/assets/1592008/3ba6a280-f015-413f-bc30-1d3c7e6df23a">


Which means the password would actually be much shorter than what the user is expecting, and actually the user doesn't even really know their own password, funny enough. I think we should make sure to handle this at some point otherwise this could come back to bite us 🤔

Reply:
Hm, ok.

But the limit here is 72 bytes (not characters), so that should be compatible with bcrypt, no?

For 2-byte characters, that means a max password length of 36 characters, which seems... fine?

OWASP recommends 72 character/byte limit: https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#:~:text=bcrypt%20has%20a%20maximum%20length,in%20use%20has%20smaller%20limits).

Reply:
The `isLength` validator does not measure bytes, see my example

Reply:
Ahh, I see.  Good call.  I'll change to 36.

Reply:
`isByteLength()` ✅

Reply:
See https://github.com/medplum/medplum/pull/3334/files#diff-902e8aa40885f63d7b3dcce2af92c11ebe5debbd9c40901a3b29ba635441cc65R137

Reply:
Wow that was an easy fix. Didn't even see that. Good find